### PR TITLE
[which] Initial packaging. Fixes JB#46906

### PR DESCRIPTION
**Pull request by Thaodan**

### CODE REVIEW

**Review by xfade — APPROVED** [review]

**Comment by okodron** [comment]
Why not using https://git.savannah.gnu.org/git/which.git as an upstream submodule?

File: which.spec [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+License: GPLv3
+Source0: http://ftp.gnu.org/gnu/which/%{name}-%{version}.tar.gz
+Source1: which2.sh
+Source2: which2.csh
+Patch0: which-2.21-coverity-fixes.patch
+Url: https://savannah.gnu.org/projects/which/
+Requires: coreutils
+BuildRequires: make
+BuildRequires: gcc
+BuildRequires: readline-devel
```
Comment (okodron) [inline which.spec]:
```suggestion
BuildRequires: pkgconfig(readline)
```

**Comment by Thaodan** [comment]
> Why not using https://git.savannah.gnu.org/git/which.git as an upstream submodule?

It is empty.

File: which.spec [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+%autosetup -p1
+
+%build
+%configure
+%make_build
+
+%install
+%make_install
+
+mkdir -p $RPM_BUILD_ROOT%{_sysconfdir}/profile.d
```
Comment (okodron) [inline which.spec]: `$RPM_BUILD_ROOT` can be replaced with `%{buildroot}`

File: which.spec [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+Version: 2.21
+Release: 1
+License: GPLv3
+Source0: http://ftp.gnu.org/gnu/which/%{name}-%{version}.tar.gz
+Source1: which2.sh
+Source2: which2.csh
+Patch0: which-2.21-coverity-fixes.patch
+Url: https://savannah.gnu.org/projects/which/
+Requires: coreutils
+BuildRequires: make
```
Comment (okodron) [inline which.spec]: Better to buildrequire `autoconf` and `libtool` here.

**Comment by okodron** [comment]
> > Why not using https://git.savannah.gnu.org/git/which.git as an upstream submodule?
> 
> It is empty.

Yeah... my bad.

File: which.spec (hunk 3 above)
Comment (Thaodan) [reply which.spec]: libtool is not used in this case, libtool is for libraries not for programs.

Comment (okodron) [reply which.spec]: Ok.

**Comment by mkosola** [comment]
Please change also .spec and .changes to gnu-which, so those are matching package name.

**Comment by Thaodan** [comment]
@mkosola: Fixed

**Comment by mkosola** [comment]
> @mkosola: Fixed

Hmm, I don't see .spec and .changes file now at all?

**Comment by Thaodan** [comment]
Forgot to add the renamed files, fixed now.

**Review by mkosola — APPROVED** [review]

**Pull request merged**